Code/Tools/ProjectManager/Source/GemRepo/GemRepoScreen.cpp

```diff
@@ -34,11 +37,136 @@ namespace O3DE::ProjectManager
         vLayout->setSpacing(0);
         setLayout(vLayout);
 
+        m_contentStack = new QStackedWidget(this);
+
+        m_noRepoContent = CreateNoReposContent();
+        m_contentStack->addWidget(m_noRepoContent);
+
+        m_repoContent = CreateReposContent();
+        m_contentStack->addWidget(m_repoContent);
+
+        vLayout->addWidget(m_contentStack);
+
+        Reinit();
+    }
+
+    void GemRepoScreen::Reinit()
+    {
+        m_gemRepoModel->clear();
+        FillModel();
+
+        // If model contains any data show the repos
+        if (m_gemRepoModel->rowCount())
+        {
+            m_contentStack->setCurrentWidget(m_repoContent);
+        }
+        else
+        {
+            m_contentStack->setCurrentWidget(m_noRepoContent);
+        }
+
+        // Select the first entry after everything got correctly sized
+        QTimer::singleShot(200, [=]{
+            QModelIndex firstModelIndex = m_gemRepoListView->model()->index(0,0);
+            m_gemRepoListView->selectionModel()->select(firstModelIndex, QItemSelectionModel::ClearAndSelect);
+        });
+    }
+
+    void GemRepoScreen::HandleAddRepoButton()
+    {
+        GemRepoAddDialog* repoAddDialog = new GemRepoAddDialog(this);
+        repoAddDialog->exec();
+
+        if (repoAddDialog->GetButtonResult() == QDialogButtonBox::ApplyRole)
+        {
+            QString repoUrl = repoAddDialog->GetRepoPath();
+            if (repoUrl.isEmpty())
+            {
+                return;
+            }
+
+            AZ::Outcome<void, AZStd::string> addGemRepoResult = PythonBindingsInterface::Get()->AddGemRepo(repoUrl);
+            if (addGemRepoResult.IsSuccess())
+            {
+                Reinit();
+            }
+            else
+            {
+                QMessageBox::critical(this, tr("Operation failed"),
+                    QString("Failed to add gem repo: %1.\nError:\n%2").arg(repoUrl, addGemRepoResult.GetError().c_str()));
```

Comment:
I could be wrong, but I think when you use \n instead of `<br>` in these message boxes, [it breaks all the auto-html formatting that might otherwise occur](https://stackoverflow.com/questions/9955827/html-in-qmessagebox).